src/main/java/homework8/ArrayValueCalculator.java

```diff
@@ -0,0 +1,63 @@
+package src.main.java.homework8;
+
+public class ArrayValueCalculator {
+
+    public static int doCalc(String[][] stringArray) throws ArraySizeException, ArrayDataException {
+
+        if (stringArray.length != 4 || stringArray[0].length != 4)
+            throw new ArraySizeException(stringArray.length, stringArray[0].length);
+
+        int sum = 0;
+
+        for (int i = 0; i < stringArray.length; i++) {
+            for (int j = 0; j < stringArray[i].length; j++) {
+                try {
+                    sum += Integer.parseInt(stringArray[i][j]);
+                } catch (Exception e) {
```

Comment:
There should be a specific error (`NumberFrmatException`) not a general `Exception` that should be wrapped into ArrayDataException.

Reply:
Why?

Reply:
Because currently, any exception that is a child of  `Exception` will be catched and treated as `ArrayDataException`. For example RuntimeExceptions, or IOExceptions, etc... any that have `Exception` as a parent. But according to the task - you have to throw ArrayDataException only if something goes wrong with parsing string to int - and this "parsing" is throwing a specific exception - `NumberFormatException`.

Reply:
Yeah, Thanks, I`ve got it. Thought that it would be better to catch any exception.